network/network-tutorial-overlay.md

```diff
@@ -328,66 +326,66 @@ The example uses Linux hosts, but the same commands work on Windows.
-        `host2`, and execute the command. Add the `--advertise-addr` flag,
-        specifying the IP address for the interface that will communicate with
-        the other host (for instance, the private IP address on AWS). The
-        last argument is the IP address of `host1`.
+    b.  On `host2`, join the swarm by running the `docker swarm join` command
+        (plus token, ip, and port) returned when you initialized on `host1`:
 
         ```bash
         (host2) $ docker swarm join \
                   --token SWMTKN-1-3mtj3k6tkuts4cpecpgjdvgj1u5jre5zwgiapox0tcjs1trqim-bfwb0ve6kf42go1rznrn0lycx \
-                  --advertise-addr 192.0.2.2:2377 \
                   192.0.2.1:2377
         ```
 
-        If the command succeeds, the following message is shown:
+    If the command succeeds, the following message is shown:
 
         ```none
         This node joined a swarm as a worker.
         ```
 
-        Otherwise, the `docker swarm join` command will time out. In this case,
-        run `docker swarm leave --force` on `node2`, verify your network and
-        firewall settings, and try again.
+    Otherwise, the `docker swarm join` command times out. If so, run `docker
+    swarm leave --force` on `host2`, verify your network and firewall
+    settings, and try again.
 
-2.  Create an attachable overlay network called `test-net` on `host1`.
+2.  On `host1`, create an attachable overlay network called `test-net`:
 
     ```bash
-    $ docker network create --driver=overlay --attachable test-net
+    (host1) $ docker network create --driver=overlay --attachable test-net
     ```
 
-    You don't need to manually create the overlay on `host2` because it will
-    be created when a container or service tries to connect to it from `host2`.
+    > On `host2`, the `test-net` overlay network is created when you run `alpine2` on it.
 
-3.  On `host1`, start a container that connects to `test-net`:
+3.  On `host1`, start an interactive container (`alpine1`), with a terminal, that connects to `test-net`:
 
     ```bash
-    (host1) $ docker run -dit \
-              --name alpine1 \
-              --network test-net \
-              alpine
+    (host1) $ docker run -it --name alpine1 --network test-net alpine
+    / #
     ```
 
-4.  On `host2`, start a container that connects to `test-net`:
+4.  On `host2`, list the available networks -- notice that `test-net` does not yet exist:
 
     ```bash
-    (host2) $ docker run -dit \
-              --name alpine2 \
-              --network test-net \
-              alpine
+    (host2) $ docker network ls
     ```
-    The `-dit` flags mean to start the container detached
-    (in the background), interactive (with the ability to type into it), and
-    with a TTY (so you can see the input and output).
-    
-    > **Note**: There is nothing to prevent you from using the same container
-    > name on multiple hosts, but automatic service discovery will not work if
```

Comment:
So; I think the confusion here around "service discovery" is basically that it's not a "service", but "dns discovery of containers", is that what you meant?

Reply:
It is now! That makes sense, @thaJeztah . Thank you.